presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -1435,6 +1443,30 @@ void PrestoServer::reportNodeStatus(proxygen::ResponseHandler* downstream) {
   http::sendOkResponse(downstream, json(fetchNodeStatus()));
 }
 
+void PrestoServer::handleGracefulShutdown(
+    const std::vector<std::unique_ptr<folly::IOBuf>>& body,
+    proxygen::ResponseHandler* downstream) {
+  if (body.size() == 1 &&
+      folly::trimWhitespace(body[0]->moveToFbString()) == http::kShuttingDown) {
+    LOG(INFO) << "Shutdown requested";
+    if (nodeState() == NodeState::kActive) {
+      // Run stop() in a separate thread to avoid blocking the main HTTP handler
+      // and ensure a timely 200 OK response to the client.
+      std::thread([this]() { this->stop(); }).detach();
+    } else {
+      LOG(INFO) << "Node is inactive or shutdown is already requested";
+    }
+    http::sendOkResponse(downstream);
+  } else {
+    std::string bodyContent;
+    for (const auto& buf : body) {
+        bodyContent+=std::string(folly::trimWhitespace(buf->clone()->moveToFbString()));
+    }
+    LOG(ERROR) << "Bad Request. Received body content: "<<bodyContent;
```

Comment:
nit: space after `<<`

Reply:
Actually this is not working right now I am getting empty line, working on a fix.

Reply:
Done.